From 18f92fea441ad6cad6723f6c60cec1449584dd2b Mon Sep 17 00:00:00 2001
From: Richard Acayan <mailingradian@gmail.com>
Date: Mon, 2 Oct 2023 20:54:45 -0400
Subject: [PATCH] arm64: dts: qcom: sdm630: add lpass digital codec

The digital codec is responsible for controlling playback volume and
sending PCM data to the analog codec. Add it.

Signed-off-by: Richard Acayan <mailingradian@gmail.com>

Source-Commit: https://github.com/sdm660-mainline/linux/commit/18f92fea441ad6cad6723f6c60cec1449584dd2b
Scope: SDM630/660-common
Porting notes: Original digital codec node hunk applies unchanged to Linux 7.3-rc5.
---
diff --git a/arch/arm64/boot/dts/qcom/sdm630.dtsi b/arch/arm64/boot/dts/qcom/sdm630.dtsi
index f1ba02913e10f2..fd6d1f3b341cad 100644
--- a/arch/arm64/boot/dts/qcom/sdm630.dtsi
+++ b/arch/arm64/boot/dts/qcom/sdm630.dtsi
@@ -16,6 +16,7 @@
 #include <dt-bindings/interrupt-controller/arm-gic.h>
 #include <dt-bindings/thermal/thermal.h>
 #include <dt-bindings/soc/qcom,apr.h>
+#include <dt-bindings/sound/qcom,q6dsp-lpass-ports.h>
 
 / {
 	interrupt-parent = <&intc>;
@@ -2417,6 +2418,23 @@
 			};
 		};
 
+		lpass_codec: audio-codec@152c0000 {
+			compatible = "qcom,msm8916-wcd-digital-codec";
+			reg = <0x152c0000 0x400>;
+
+			clocks = <&int_cdc_bclk>,
+				<&q6afecc LPASS_CLK_ID_INT_MCLK_0 LPASS_CLK_ATTRIBUTE_COUPLE_NO>;
+			clock-names = "ahbix-clk", "mclk";
+
+			assigned-clocks = <&q6afecc LPASS_CLK_ID_INT_MCLK_0 LPASS_CLK_ATTRIBUTE_COUPLE_NO>;
+			assigned-clock-rates = <9600000>;
+			// Resolve naming conflicts with digital and analog PDM_RX*
+			sound-name-prefix = "Digital";
+			#sound-dai-cells = <1>;
+
+			status = "disabled";
+		};
+
 		adsp_pil: remoteproc@15700000 {
 			compatible = "qcom,sdm660-adsp-pas";
 			reg = <0x15700000 0x4040>;

